### ISSUE TITLE
TimeInput - hogging focus

### ISSUE THREAD

**Issue by jcfilben** [grommet/grommet#8083]
When the input within TimeInput has focus and I click on another item on the page my focus keeps jumping back to TimeInput.

Noticed this behavior in this story: https://storybook.grommet.io/?path=/story/input-timeinput--time-form

The video shows me trying to move my focus to the storybook search input but being unable to do so.
https://github.com/user-attachments/assets/65f20f69-72d0-42bc-9183-8c563d9447f0
